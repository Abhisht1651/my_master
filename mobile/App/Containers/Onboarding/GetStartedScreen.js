// @flow

import { NavigationContainerProps } from '@react-navigation/native'

import React from 'react'
import { Text, Image, View, Platform } from 'react-native'
import { Images } from '../../Themes'

// Styles
import styles from '../Styles/Onboarding/GetStartedScreenStyles'
import { MediumButton } from '../../Components/Buttons/MediumButton'

type Props = NavigationContainerProps & {}

const GetStarted = (props: Props) => {
  return (
    <View style={styles.mainContainer}>
      {Platform.OS === 'ios' && <View style={styles.statusBar} />}
      <View style={styles.section}>
        <Image source={Images.newzeraPurpleLogo} style={styles.logo} />
      </View>
      <View style={styles.slogan}>
        <Text style={styles.sloganText}>A better way of catching up.</Text>
      </View>
      <View style={styles.buttonWrap}>
        <MediumButton title="Get Started" onPress={() => props.navigation.navigate('SignUp')} />
        <View style={styles.loginButton}>
          <MediumButton
            outline
            title="Log In"
            onPress={() => props.navigation.navigate('LoginScreen')}
          />
        </View>
      </View>
    </View>
  )
}

export default GetStarted
