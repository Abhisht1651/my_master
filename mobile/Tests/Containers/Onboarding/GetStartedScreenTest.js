import { Image, Text } from 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import GetStartedScreen from '../../../App/Containers/Onboarding/GetStartedScreen'
import { MediumButton } from '../../../App/Components/Buttons/MediumButton'

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn()
  },
  ...props
})
test('GetStartedScreen renders correctly', () => {
  const props = createTestProps({})
  const tree = renderer.create(<GetStartedScreen {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
describe('GetStartedScreen', () => {
  describe('rendering', () => {
    let wrapper
    let props
    beforeEach(() => {
      props = createTestProps({})
      wrapper = shallow(<GetStartedScreen {...props} />) // no compile-time error
    })

    it('should render a <Image />', () => {
      expect(wrapper.find(Image)).toHaveLength(1)
    })
    it('should render <Text /> tag', () => {
      expect(wrapper.find(Text)).toHaveLength(1)
    })
    it('should render 2 <Rounded Button />', () => {
      expect(wrapper.find(MediumButton)).toHaveLength(2)
    })

    it('should navigate to SignUp Screen when clicked on Get Started Button', () => {
      const button1 = wrapper.find(MediumButton).at(0)
      button1.simulate('press')
      expect(props.navigation.navigate).toHaveBeenCalledWith('SignUp')
    })
    it('should navigate to LogIn Screen when clicked on Log In', () => {
      const button1 = wrapper.find(MediumButton).at(1)
      button1.simulate('press')
      expect(props.navigation.navigate).toHaveBeenCalledWith('LoginScreen')
    })
  })
})
