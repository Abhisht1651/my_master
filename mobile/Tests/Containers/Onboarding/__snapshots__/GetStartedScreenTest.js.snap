// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStartedScreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 58.188802083333336,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "height": 226.78000000000003,
          "marginTop": 237.45200000000003,
          "resizeMode": "contain",
          "width": 434.99999999999994,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": "2.5%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Medium",
          "fontSize": 38,
        }
      }
    >
      A better way of catching up.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 295.3476,
      }
    }
  >
    <View
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#7B46E4",
              "borderRadius": 78,
              "height": 120,
              "width": 550,
            },
          ],
          Object {},
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "BananaGrotesk-Bold",
                  "fontSize": 32,
                  "letterSpacing": -0.18,
                  "lineHeight": 36,
                },
              ],
              Object {},
            ]
          }
        >
          Get Started
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 17.342000000000002,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#7B46E4",
              "borderRadius": 78,
              "borderWidth": 2,
              "height": 120,
              "width": 550,
            },
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#7B46E4",
                  "fontFamily": "BananaGrotesk-Bold",
                  "fontSize": 32,
                  "letterSpacing": -0.18,
                  "lineHeight": 36,
                },
                Object {},
              ]
            }
          >
            Log In
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
